packages/core/src/components/Modal/private/InnerContent.tsx

```diff
@@ -11,40 +12,151 @@ export type Props = {
   children: NonNullable<React.ReactNode>;
   /** Footer content. */
   footer?: React.ReactNode;
-  /** True to show the large version of the Dialog. */
+  /** Show the large version of the Dialog. */
   large?: boolean;
+  /** Show the medium version of the Dialog. */
+  medium?: boolean;
+  /** Modal content height becomes scrollable. */
+  scrollable?: boolean;
+  /** Dialog header subtitle. */
+  subtitle?: React.ReactNode;
   /** Dialog header title. */
   title?: React.ReactNode;
   /** Callback for when the Dialog should be closed.  */
   onClose: (event: React.MouseEvent<any> | React.KeyboardEvent) => void;
 };
 
 /** A Dialog component with a backdrop and a standardized layout. */
-export default function ModalInnerContent({ children, footer, large, onClose, title }: Props) {
+function ModalInnerContent({
+  children,
+  cx,
+  footer,
+  large,
+  medium,
+  onClose,
+  scrollable,
+  styles,
+  subtitle,
+  title,
+}: Props & WithStylesProps) {
   const theme = useTheme();
+  const withHeader = Boolean(title || subtitle);
+  const withFooter = Boolean(footer);
 
   return (
-    <Spacing bottom={6} horizontal={4} top={4}>
-      <Spacing bottom={title ? 3 : 0} tag="header">
-        <Spacing bottom={4}>
-          <IconButton onClick={onClose}>
-            <IconClose
-              accessibilityLabel={T.phrase('Close', {}, 'Close a modal popup')}
-              size={3 * theme.unit}
-            />
-          </IconButton>
-        </Spacing>
-
-        {title && <Title level={large ? 1 : 3}>{title}</Title>}
-      </Spacing>
-
-      {children}
+    <div className={cx(styles.wrapper)}>
+      {withHeader && (
+        <header className={cx(styles.header, scrollable && styles.header_scrollable)}>
+          {title && <Title level={3}>{title}</Title>}
+          {subtitle && <Text muted>{subtitle}</Text>}
+        </header>
+      )}
+
+      <div className={cx(styles.close, !withHeader && styles.close_float)}>
+        <IconButton onClick={onClose}>
+          <IconClose
+            accessibilityLabel={T.phrase('Close', {}, 'Close a modal popup')}
+            color={theme.color.muted}
+            size={theme.unit * 3}
+          />
+        </IconButton>
+      </div>
+
+      <div
+        className={cx(
+          styles.body,
+          !withHeader && styles.body_paddingTop,
```

Comment:
Maybe the body always has the padding and header/footer have the margin 0?

Reply:
i chose to keep it separate because of the footer with its border when scrollable and when the header is not there. and when scrollable is active, separate padding concerns for the body anyways. basically, found it easier to keep header/footer padding to themselves and add it in the cases when they are not present and the body could use it

Reply:
Makes sense 👍